<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Estimate Genome Size of Polyploid Species Using K-mer Frequencies</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Estimate Genome Size of Polyploid Species
Using K-mer Frequencies</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(findGSEP)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co">#&gt; Loading required package: RColorBrewer</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#&gt; Loading required package: ggplot2</span></span></code></pre></div>
<div id="description" class="section level2">
<h2>Description</h2>
<p><code>findGSEP</code> is a function for multiple polyploidy genome
size estimation by fitting k-mer frequencies iteratively with a normal
distribution model.</p>
<p>To use <code>findGSEP</code>, one needs to prepare a histo file,
which contains two tab-separated columns. The first column gives
frequencies at which k-mers occur in reads, while the second column
gives counts of such distinct k-mers. Parameters k and related histo
file are required for any estimation.</p>
<p>Dependencies (R library) required: <code>pracma</code>,
<code>fGarch</code>, etc. - see DESCRIPTION for details.</p>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">findGSEP</span>(</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  path,</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  samples,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  sizek,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  exp_hom,</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  ploidy,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  range_left,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  range_right,</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  xlimit,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  ylimit,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  output_dir</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="arguments" class="section level2">
<h2>Arguments</h2>
<ul>
<li><p><strong>path</strong>: is the histo file location
(mandatory).</p></li>
<li><p><strong>samples</strong>: is the histo file name
(mandatory).</p></li>
<li><p><strong>sizek</strong>: is the size of k used to generate the
histo file (mandatory). K is involved in calculating heterozygosity if
the genome is heterozygous.</p></li>
<li><p><strong>exp_hom</strong>: a rough average k-mer coverage for
finding the homozygous regions. In general, one can get peaks in the
k-mer frequencies file, but has to determine which one is for the
homozygous regions, and which one is for the heterozygous regions. It is
optional, however, it must be provided if one wants to estimate size for
a heterozygous genome. VALUE for <code>exp_hom</code> must satisfy
<code>fp &lt; VALUE &lt; 2*fp</code>, where <code>fp</code> is the freq
for homozygous peak. If not provided, 0 by default assumes the genome is
homozygous.</p></li>
<li><p><strong>ploidy</strong>: is the number of ploidy
(mandatory).</p></li>
<li><p><strong>range_left</strong>: is the left range for estimation,
default is <code>exp_hom * 0.2</code>, normally do not need to change
this (optional).</p></li>
<li><p><strong>range_right</strong>: is the right range for estimation,
default is <code>exp_hom * 0.2</code>, normally do not need to change
this (optional).</p></li>
<li><p><strong>xlimit</strong>: is the x-axis range, if not given, then
it will automatically calculate a proper range, normally do not need to
change this (optional).</p></li>
<li><p><strong>ylimit</strong>: is the y-axis range, if not given, then
it will automatically calculate a proper range, normally do not need to
change this (optional).</p></li>
<li><p><strong>output_dir</strong>: is the path to write output files
(optional). If not provided, by default results will be written in the
folder where the histo file is.</p></li>
</ul>
</div>
<div id="example-usage" class="section level2">
<h2>Example Usage</h2>
<p>To run the algorithm, follow these steps:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Prepare a Path</strong>: Create a directory where the
histo file will be stored. For example, create a directory named
<code>test_findGSEP</code>.</p></li>
<li><p><strong>Put Histo File in the Path</strong>: Place your histo
file in the <code>test_findGSEP</code> directory. In this example, the
histo file name is
<code>ara_simulate_4ploidy_25x_rep4.histo</code>.</p></li>
<li><p><strong>Provide Output Directory</strong>: Specify the output
directory where the results will be saved. In this example, we use
<code>tempdir()</code> as the output directory.</p></li>
<li><p><strong>Run the Algorithm</strong>: Use the following command to
run the algorithm with the specified parameters:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">findGSEP</span>(</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    <span class="at">path =</span> <span class="st">&#39;test_findGSEP&#39;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="at">samples =</span> <span class="st">&#39;ara_simulate_4ploidy_25x_rep4.histo&#39;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>    <span class="at">sizek =</span> <span class="dv">21</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="at">exp_hom =</span> <span class="dv">35</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="at">ploidy =</span> <span class="dv">4</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    <span class="at">output_dir =</span> <span class="fu">tempdir</span>(),</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="at">range_left =</span> <span class="dv">35</span> <span class="sc">*</span> <span class="fl">0.2</span>, <span class="do">## exp_hom*0.2</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="at">range_right =</span> <span class="dv">35</span> <span class="sc">*</span> <span class="fl">0.2</span>, <span class="do">## exp_hom*0.2</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>    <span class="at">xlimit =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="do">## will calculate automatically</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>    <span class="at">ylimit =</span> <span class="sc">-</span><span class="dv">1</span> <span class="do">## will calculate automatically</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>)</span></code></pre></div></li>
<li><p><strong>Output</strong>: The output will include:</p>
<ul>
<li>A PDF file named <code>${samples}._hap_genome_size_est.pdf</code>,
which contains the estimated genome size.</li>
<li>A CSV file named <code>${samples}._haploid_size.csv</code>, which
contains the predicted genome size.</li>
</ul></li>
</ol>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Laiyi Fu, Yanxin Xie, Shunkang Ling, and Hequan Sun# etc.
al.Â findGSEP: a web application for estimating ge-nome size of polyploid
species using k-mer frequencies</p>
</div>
<div id="session-info" class="section level2">
<h2>Session Info</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">3.3</span> (<span class="dv">2024-02-29</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin20</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="dv">14</span>.<span class="fl">4.1</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.3</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">11.0</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>time zone<span class="sc">:</span> Asia<span class="sc">/</span>Shanghai</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>[<span class="dv">1</span>] findGSEP_1.<span class="fl">2.0</span>     dplyr_1.<span class="fl">1.4</span>        png_0<span class="fl">.1</span><span class="dv">-8</span>          scales_1.<span class="fl">3.0</span>       fGarch_4033<span class="fl">.92</span>    </span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>[<span class="dv">6</span>] pracma_2.<span class="fl">4.4</span>       ggplot2_3.<span class="fl">5.0</span>      RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a> [<span class="dv">1</span>] Matrix_1<span class="fl">.6</span><span class="dv">-5</span>        gtable_0.<span class="fl">3.5</span>        compiler_4.<span class="fl">3.3</span>      fBasics_4032<span class="fl">.96</span>     gbutils_0<span class="fl">.5</span>        </span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a> [<span class="dv">6</span>] tidyselect_1.<span class="fl">2.1</span>    cvar_0<span class="fl">.5</span>            timeSeries_4032<span class="fl">.109</span> yaml_2.<span class="fl">3.8</span>          fastmap_1.<span class="fl">1.1</span>      </span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>[<span class="dv">11</span>] lattice_0<span class="fl">.22</span><span class="dv">-5</span>      R6_2.<span class="fl">5.1</span>            generics_0.<span class="fl">1.3</span>      knitr_1<span class="fl">.45</span>          rbibutils_2.<span class="fl">2.16</span>   </span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>[<span class="dv">16</span>] tibble_3.<span class="fl">2.1</span>        spatial_7<span class="fl">.3</span><span class="dv">-17</span>      munsell_0.<span class="fl">5.1</span>       timeDate_4032<span class="fl">.109</span>   pillar_1.<span class="fl">9.0</span>       </span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>[<span class="dv">21</span>] rlang_1.<span class="fl">1.3</span>         utf8_1.<span class="fl">2.4</span>          xfun_0<span class="fl">.43</span>           pkgload_1.<span class="fl">3.4</span>       cli_3.<span class="fl">6.2</span>          </span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>[<span class="dv">26</span>] withr_3.<span class="fl">0.0</span>         magrittr_2.<span class="fl">0.3</span>      Rdpack_2<span class="fl">.6</span>          digest_0.<span class="fl">6.35</span>       grid_4.<span class="fl">3.3</span>         </span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>[<span class="dv">31</span>] rstudioapi_0.<span class="fl">16.0</span>   lifecycle_1.<span class="fl">0.4</span>     vctrs_0.<span class="fl">6.5</span>         evaluate_0<span class="fl">.23</span>       glue_1.<span class="fl">7.0</span>         </span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>[<span class="dv">36</span>] fansi_1.<span class="fl">0.6</span>         colorspace_2<span class="fl">.1</span><span class="dv">-0</span>    rmarkdown_2<span class="fl">.26</span>      htmltools_0.<span class="dv">5</span>.<span class="fl">8.1</span>   tools_4.<span class="fl">3.3</span>        </span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>[<span class="dv">41</span>] pkgconfig_2.<span class="fl">0.3</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
